staging/src/k8s.io/client-go/tools/cache/controller.go

```diff
@@ -351,7 +351,38 @@ func NewIndexerInformer(
 	// This will hold the client state, as we know it.
 	clientState := NewIndexer(DeletionHandlingMetaNamespaceKeyFunc, indexers)
 
-	return clientState, newInformer(lw, objType, resyncPeriod, h, clientState)
+	return clientState, newInformer(lw, objType, resyncPeriod, h, clientState, nil)
+}
+
+// TransformFunc allows for transforming an object before it will be processed
```

Comment:
@liggitt - obviously tests are missing in this PR, but before I go ahead and add them I would appreciate some early feedback on this change.

Reply:
@deads2k - would you mind providing some early feedback?

Reply:
I'm ok with the idea, but I'd like to be sure that the layers above this one make it difficult to accidentally confuse your listers with different transformations.

Reply:
> make it difficult to accidentally confuse your listers with different transformations.

Can you clarify? Is the comment about naming mostly? If so, I'm happy to change if you have better suggestions.

Reply:
> If so, this effectively would be the same interface, so it would be possible once to another, so I'm not sure it would really bring value. WDYT?

Duck typing never fails to disappoint.  I think you're right.